src/main/java/in/twizmwaz/cardinal/module/apply/AppliedModule.java

```diff
@@ -0,0 +1,258 @@
+                      "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+              continue;
+            }
+          }
+
+          if (type.filterOnly) {
+            applied.get(match).add(new AppliedRegion(type, region, filter, message, earlyWarning));
+          } else {
+            if (filter == null) {
+              //TODO: always filter doesn't exist yet
+              filter = filterModule.getFilter(match, "always");
+            }
+            if (type.equals(ApplyType.VELOCITY)) {
+              Vector vel = Numbers.getVector(applyElement.getAttributeValue(type.otherAttr));
+              if (vel == null) {
+                Located attrLocated = (Located) applyElement.getAttribute(type.otherAttr);
+                errors.add(new ModuleError(this, match.getMap(),
+                    new String[]{"Invalid vector specified for applied region attribute: " + type.otherAttr,
+                        "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+                continue;
+              }
+              applied.get(match).add(new AppliedRegion(region, filter, vel));
+            } else {
+              Kit kit = kitModule.getKit(match, applyElement.getAttributeValue(type.otherAttr));
+              if (kit == null) {
+                Located attrLocated = (Located) applyElement.getAttribute(type.otherAttr);
+                errors.add(new ModuleError(this, match.getMap(),
+                    new String[]{"Invalid kit specified for applied region attribute: " + type.otherAttr,
+                        "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+                continue;
+              }
+              applied.get(match).add(new AppliedRegion(type, region, filter, kit));
+            }
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  public void add(Match match, AppliedRegion appliedRegion) {
+    add(match, appliedRegion, false);
+  }
+
+  public void add(Match match, AppliedRegion appliedRegion, boolean prioritize) {
+    applied.get(match).add(prioritize ? 0 : applied.get(match).size(), appliedRegion);
+  }
+
+  public List<AppliedRegion> get(Match match, ApplyType... types) {
+    List<ApplyType> typeList = Arrays.asList(types);
+    return applied.get(match).stream().filter(appliedRegion ->
+        typeList.contains(appliedRegion.getType())).collect(Collectors.toList());
+  }
+
+  //TODO: Regions shouldn't apply to observers
+  @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+  public void onPlayerMove(PlayerMoveEvent event) {
+    Vector from = event.getFrom().toVector();
+    Vector to = event.getTo().toVector();
+    
+    for (AppliedRegion reg : get(Cardinal.getMatch(event.getPlayer()), ApplyType.ENTER, ApplyType.LEAVE)) {
+
+      if ((reg.isType(ApplyType.ENTER) && !reg.contains(from) && reg.contains(to))
```

Comment:
You have an if enclosing an if. Java evaluates conditions in order, and if the result of the statement can be determined without evaluating more conditions, it will proceed.